azure-kusto-ingest/azure/kusto/ingest/descriptors.py

```diff
@@ -20,6 +22,10 @@ def assert_uuid4(maybe_uuid, error_message):
 class FileDescriptor:
     """FileDescriptor is used to describe a file that will be used as an ingestion source."""
 
+    # The compressed sized of a 4GB file that will not cause the gzip stream size (unsigned int) to rollover given a 1:40 compression factor
```

Comment:
not sure I understand this. could you please try and rephrase the comment? I understand there is a max size for a single gzip file to store it's uncompressed size, but I don't understand what it is, and why we are assuming a 1:40 compression rate

Reply:
Let me know what you think 

    # Gzip keeps the decompressed stream size as a UINT32 in the last 4 bytes of the stream, however this poses a limit to the expressed size which is 4GB
    # The standard says that when the size is bigger then 4GB, the UINT rolls over.
    # The below constant expresses the maximal size of a compressed stream that will not cause the UINT32 to rollover given a maximal compression ratio of 1:40

Reply:
much better :) thanks!